File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minTimeToVisitAllPoints = function(p) {
    let min = 0;
    for (let i = 0; i < p.length - 1; i++) {
        let x = p[i][0]
        let y = p[i][1]
        let X = p[i + 1][0]
        let Y = p[i + 1][1]

        min += Math.max(Math.abs(X - x), Math.
        abs(Y - y));
    }
    return min;
};

File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.java
class Solution {
    public int minTimeToVisitAllPoints(int[][] p) {
        int min = 0;
        for (int i = 0; i < p.length - 1; i++) {
            int x = p[i][0];
            int y = p[i][1];
            int X = p[i + 1][0];
            int Y = p[i + 1][1];
            min += Math.max(Math.abs(X - x), Math.abs(Y - y));
        }

        return min;
    }
}